include/xgboost/collective/poll_utils.h

```diff
@@ -35,7 +35,12 @@
 
 #if !defined(_WIN32)
 
+#include "xgboost/unixdefs.h"
+#ifdef __MUSL__
+#include <poll.h>
+#else
 #include <sys/poll.h>
```

Comment:
Apologies for the wrong import. I think `#include <poll.h>` should work fine given https://man7.org/linux/man-pages/man2/poll.2.html